### PR TITLE
Us33

### PR DESCRIPTION
**Pull request by ross-ian28**
## Description of Changes
Merchant Items show page has a link to update the items attributes
## Concerns (tags recommended)
None
## What gif describes how you feel about this PR?
ferret_lick_lips

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -3,7 +3,7 @@
   resources :merchants do
 
     resources :invoices, controller: 'merchant_invoices'
-    resources :items, controller: 'merchant_items', only: [:index, :show]
+    resources :items, controller: 'merchant_items', only: [:index, :show, :edit, :update]
```
Comment (glsorensen) [inline config/routes.rb:6]: Nice job adding on the routes you need to the resources.

File: spec/features/merchant_items/show_spec.rb [hunk 2]
```diff
@@ -26,3 +26,39 @@
...
+  before(:each) do
+    @merchant1 = Merchant.create!(name: "Pabu")
+    @merchant2 = Merchant.create!(name: "Loki")
+    @item1 = @merchant1.items.create!(name: "Ferret Food", description: "Yummy", unit_price: 20)
+    @item2 = @merchant1.items.create!(name: "Ferret Leash", description: "Long", unit_price: 20)
+    @item3 = @merchant2.items.create!(name: "Ferret Shampoo", description: "Tastey", unit_price: 20)
+  end
+
+  it 'has link to take to edit page' do
+    visit "/merchants/#{@merchant1.id}/items/#{@item1.id}"
```
Comment (glsorensen) [inline spec/features/merchant_items/show_spec.rb:40]: Can you do this by referencing the path with advanced routing?

File: app/controllers/merchant_items_controller.rb [hunk 3]
```diff
@@ -22,7 +23,17 @@ def show
...
+    @item = Item.find(params[:id])
+    @item.update({
+      name: params[:name],
+      description: params[:description],
+      unit_price: params[:unit_price]
+      })
+
+    redirect_to "/merchants/#{@merchant.id}/items/#{@item.id}"
 
+    flash[:alert] = "Item has been updated"
```
Comment (gregoryjflaherty) [inline app/controllers/merchant_items_controller.rb:36]: Love the flash alert

**Review by glsorensen — APPROVED** [review]
I pulled the branch to my local and all tests were passing, looks good to me. Just check if you can update the route references in your spec to advanced routing.

File: config/routes.rb (hunk 1 above)
Comment (gregoryjflaherty) [inline config/routes.rb:6]: At this point we might want to move to an , except[:destroy, etc.]

**Review by gregoryjflaherty — APPROVED** [review]

File: app/controllers/merchant_items_controller.rb (hunk 3 above)
Comment (jgrazulis) [reply app/controllers/merchant_items_controller.rb:36]: Haha same-- very direct

File: config/routes.rb (hunk 1 above)
Comment (jgrazulis) [reply config/routes.rb:6]: Yea if we're using basically everything else that might be a good idea

File: spec/features/merchant_items/show_spec.rb (hunk 2 above)
Comment (jgrazulis) [reply spec/features/merchant_items/show_spec.rb:40]: Yea, if possible that sounds like a great edit

File: app/views/merchant_items/edit.html.erb [hunk 4]
```diff
@@ -0,0 +1,13 @@
+<h1>Edit Item</h1>
+<%= form_with url: "/merchants/#{@merchant.id}/items/#{@item.id}", method: :patch, local: true do |f| %>
```
Comment (jgrazulis) [inline app/views/merchant_items/edit.html.erb:2]: I'm not sure if this is a thing, but is this a spot where we could use advanced routing too?

**Pull request merged**